crates/uv-auth/src/middleware.rs

```diff
@@ -48,6 +49,54 @@ impl NetrcMode {
     }
 }
 
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq)]
+pub enum AuthPolicy {
+    /// Attempt to authenticate if needed.
+    #[default]
+    Auto,
+    /// Always attempt to authenticate.
+    Always,
+    /// Never attempt to authenticate.
+    Never,
+}
+
+#[derive(Debug, Default, Clone, Eq, PartialEq)]
+pub struct UrlAuthPolicies(FxHashMap<Url, AuthPolicy>);
+
+impl UrlAuthPolicies {
+    pub fn new() -> Self {
+        Self(FxHashMap::default())
+    }
+
+    /// Create a new `UrlAuthPolicies` from a list of URL and `AuthPolicy`
+    /// tuples.
+    pub fn from_tuples(tuples: &[(Url, AuthPolicy)]) -> Self {
+        let mut auth_policies = UrlAuthPolicies::new();
+        for (url, auth_policy) in tuples {
+            auth_policies.add_policy(url.clone(), *auth_policy);
+        }
+        auth_policies
+    }
+
+    /// An an `AuthPolicy` for a URL.
+    pub fn add_policy(&mut self, url: Url, auth_policy: AuthPolicy) {
+        self.0.insert(url, auth_policy);
+    }
+
+    /// Get the `AuthPolicy` for a URL.
+    pub fn policy_for(&self, url: &Url) -> AuthPolicy {
+        // TODO: There are probably not many URLs to iterate through,
```

Comment:
We tend to use the `TODO(name)` style in uv, where `name` indicates the author (i.e., the individual with context on the TODO -- not necessarily the individual response for resolving it).